draft-leon-distributed-multi-signer-00.md

```diff
@@ -444,45 +484,155 @@ has been done, the local MSA is able to initiate communication with
 the remote MSA and verify the identity of the responding party via the
 TLSA record for the remote MSAs certificate.
 
-## Locating a Remote DNS-Method Multi-Signer Agent
+## Multi-Signer EDNS(0) Option Format 
 
-Locating a remote MSA using the DNS mechanism consists of the
-following steps:
+This document uses an Extended Mechanism for DNS (EDNS0) {{!RFC6891}}
+option to include Key State information in DNS messages. The option is 
+structured as follows: 
 
- * Lookup and DNSSEC-validate the URI record of the MSIGNER
-   identity. This provides the domain name and port to which DNS
-   messages should be sent.
- * Lookup and DNSSEC-validate the KEY record of the URI record target
-   name. This enables verification of the SIG(0) public key of the
-   remote MSA once communication starts.
+~~~
+                                               1   1   1   1   1   1 
+       0   1   2   3   4   5   6   7   8   9   0   1   2   3   4   5 
+     +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ 0:  |                            OPTION-CODE                        |
+     +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ 2:  |                           OPTION-LENGTH                       |
+     +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ 4:  |           OPERATION           |           TRANSPORT           |
+     +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+ 8:  |        SYNCHRONIZATION        |                               /
+     +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+ 
+ 10: / OPERATION-BODY                                                /
+     +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+~~~
 
-Example: given the following MSIGNER record for a remote MSA:
+Field definition details: 
 
-zone.example.     IN MSIGNER ON  DNS msa.provider.com.
+OPTION-CODE: 
+    2 octets / 16 bits (defined in {{!RFC6891}}) contains the value TBD
+    for KeyState.
 
-The local MSA will look up the URI record for msa.provider.com:
+OPTION-LENGTH: 
+    2 octets / 16 bits (defined in {{!RFC6891}}) contains the length of 
+    the payload (everything after OPTION-LENGTH) in octets and should 
+    be 3 plus the length of the EXTRA-TEXT field (which may be zero 
+    octets long). 
 
-_dns._tcp.msa.provider.com.  IN  URI  10 10 “dns://ns.msa.provider.com:5399/”
-_dns._tcp.msa.provider.com.  IN  RRSIG URI …
+OPERATION:
+    8 bits. Signals the type of operation the message performs. Currently,
+    the operations HELLO and HEARTBET exist.
```

Comment:
This document defines the two operations HELLO and HEARTBEAT. For a complete distributed multi-signer specification a number of additional operations will need to be specified, either in a revision to this document or in a subsequent document.